spanner/api/Program.cs

```diff
@@ -403,9 +399,13 @@ AlbumTitle   STRING(MAX)
             using (TransactionScope scope = new TransactionScope(
                 TransactionScopeAsyncFlowOption.Enabled))
             {
-                // Create connection to spanner.
-                using (var connection = new SpannerConnection(
-                    connectionString))
+                // Create connection to Cloud Spanner.
+                var connection = new SpannerConnection(
+                    connectionString);
+                // Open connection as read only within the scope 
+                // of the current transaction.
+                connection.OpenAsReadOnly();
+                using (connection)

```

Comment:
you should do the using when you declare it -- as:

using (var connection = new SpannerConnection(connectionString)) {

    await connection.OpenAsReadOnlyAsync().ConfigureAwait(false);